src/extract/contextualized_data_frame.rs

```diff
@@ -155,6 +156,40 @@ impl ContextualizedDataFrame {
             .filter_map(|sc| sc.get_building_block_id())
             .collect()
     }
+
+    pub fn get_subject_id_col(&self) -> &Column {
+        self.filter_columns()
+            .where_header_context(Filter::Is(&Context::None))
+            .where_data_context(Filter::Is(&Context::SubjectId))
+            .collect()[0]
+    }
+
+    /// Given a column of String datatype in a ContextualisedDataFrame, this function will create a patient_ID -> data HashMap.
+    /// Where the data is whatever is contained in the cells of the string column.
+    ///
+    /// NOTE: no validation is done to ensure that there is exactly one value in the data for each patient.
+    /// If there are different cell values for the same patient, the value with the greatest row index will be chosen as the value in the HashMap for the patient.
+    pub fn create_subject_id_string_data_hash_map(
+        &self,
+        string_col_name: &str,
+    ) -> Result<HashMap<String, String>, PolarsError> {
```

Comment:
Try to make the type of the value of the hashmap generic.


```suggestion
    pub fn create_subject_id_string_data_hash_map<T>(
        &self,
        string_col_name: &str,
    ) -> Result<HashMap<String, T>, PolarsError> {
```

Reply:
1. I don't think the generic option will work because we are using the .str() function. I can't think of any solution that isn't a lot of effort (and solving a problem that we don't have yet)

2.

> Dont we do something very similar in the collector?

It's somewhat similar to`collect_single_multiplicity_element`. There are too many differences though for me to think of a good way to extract common features. The function `collect_single_multiplicity_element` relies on the fact that the CDF corresponds to a single patient. It also returns a single value as opposed to a hash map.